test/test_models.py

```diff
@@ -29,6 +30,47 @@ def get_models_from_module(module):
     ]
 
 
+test_vit_conv_stem_configs = [
+    models.vision_transformer.ConvStemConfig(kernel_size=3, stride=2, out_channels=64, padding=1),
+    models.vision_transformer.ConvStemConfig(kernel_size=3, stride=2, out_channels=128, padding=1),
+    models.vision_transformer.ConvStemConfig(kernel_size=3, stride=1, out_channels=128, padding=1),
+    models.vision_transformer.ConvStemConfig(kernel_size=3, stride=2, out_channels=256, padding=1),
+    models.vision_transformer.ConvStemConfig(kernel_size=3, stride=1, out_channels=256, padding=1),
+    models.vision_transformer.ConvStemConfig(kernel_size=3, stride=2, out_channels=512, padding=1),
+]
+
+
+def vitc_b_16(**kwargs: Any):
+    return models.VisionTransformer(
+        image_size=224,
+        patch_size=16,
+        num_layers=12,
+        num_heads=12,
+        hidden_dim=768,
+        mlp_dim=3072,
+        conv_stem_configs=test_vit_conv_stem_configs,
+        **kwargs,
+    )
+
+
+def vitc_l_16(**kwargs: Any):
+    return models.VisionTransformer(
+        image_size=224,
+        patch_size=16,
+        num_layers=24,
+        num_heads=16,
+        hidden_dim=1024,
+        mlp_dim=4096,
+        conv_stem_configs=test_vit_conv_stem_configs,
+        **kwargs,
+    )
+
+
+def get_models_from_builder():
+    # This function returns model builders created in this file
+    return [vitc_b_16, vitc_l_16]
+
+
```

Comment:
Nit:

Can you create a new test called `test_vitc_model(dev)` and move those definitions as nested methods within the test? You can then loop through the available methods `[vitc_b_16, vitc_l_16]` and call directly `test_classification_model(model_fn, dev)`. This way you will reuse the infra but keep things contained.